src/yews/models/focal_mechanism.py

```diff
@@ -0,0 +1,331 @@
+import torch.nn as nn
```

Comment:
Did not check the correctness here. Since you are working on the model, you konw better than me.